presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java

```diff
@@ -263,6 +268,42 @@ public boolean isFinalQueryInfo()
     }
 
     @JsonProperty
+    public String getHumanReadableState()
```

Comment:
Why is this in QueryInfo? It presentation-specific logic, so it probably belongs in the UI or, worst case, in the jaxrs resource.